src/main/java/ui/components/pickers/MilestonePicker.java

```diff
@@ -0,0 +1,53 @@
+package ui.components.pickers;
+
+import backend.resource.TurboIssue;
+import backend.resource.TurboMilestone;
+import javafx.application.Platform;
+import javafx.scene.control.ButtonType;
+import javafx.stage.Stage;
+import ui.UI;
+import undo.actions.ChangeMilestoneAction;
+import util.events.ShowMilestonePickerEventHandler;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class is meant to facilitate creating a dialog when needed for the user to set milestone for an issue
```

Comment:
Rather than the dialog itself, it's just the class that registers the handler to open the dialog, as well as trigger the necessary action according to the dialog's result.